.github/workflows/release.yaml

```diff
@@ -59,10 +59,46 @@ jobs:
           NPM_PACKAGE_ROOT: "npm"
           SKIP_DOCKER_PUBLISH: true
 
+  docs:
+    name: Update documentation
+    needs:
+      - release
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: "Generate token"
+        id: generate_token
+        uses: tibdex/github-app-token@v2
+        with:
+          app_id: ${{ secrets.OS_GITHUB_APP_ID }}
+          private_key: ${{ secrets.OS_GITHUB_APP_PRIVATE_KEY }}
+
+      - name: "☁️ checkout repository"
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ steps.generate_token.outputs.token }}
+
+      - name: "🐹 Setup Go"
+        uses: actions/setup-go@v5
+        with:
+          go-version: 1.22.x
+
+      - name: "🤲 Setup Just"
+        uses: extractions/setup-just@v2
+
+      - name: "📗 Generate Documentation"
+        run: ./scripts/generate-docs.sh
+        env:
+          GITHUB_REF: ${{ github.ref }}
+          GH_TOKEN: ${{ steps.generate_token.outputs.token }}
```

Comment:
`GH_TOKEN` doesn't actually get used in `scripts/generate-docs.sh`: shouldn't there be an extra step to authenticate with the token to git? I'm not sure what `semantic-release` is doing under the hood with it's `GITHUB_TOKEN` but I'd imagine we need to do something explicitly with it like:

```sh
git remote set-url origin "https://${GH_TOKEN}@github.com/open-sauced/pizza-cli"
```

Reply:
Gonna take a peek at this in the AM.

Reply:
> For the GH_TOKEN, I thought it was used under the hood

Ahhhhh that makes sense. Let's just send it: so hard to actually debug this stuff without seeing it in the actual action going through.